### DISCUSSION TITLE
Redact or Blur area

### DISCUSSION THREAD

**Discussion by sandstrom** [libvips/libvips#2267] [Q&A]
I'd like to blur or redact an area of an image using libvips.

There is an example here:

```python
image = pyvips.Image.new...

target_width = image.width * 2
target_height = image.width * 2
corner = image.crop(0, 0, 100, 100)
background = corner \
    .replicate(1 + target_width / 100, 1 + target_height / 100) \
    .crop(0, 0, target_width, target_height) \
    .gaussblur(3)
result = background.insert(image, (target_width - image.width) / 2, (target_height - image.height) / 2)
```

Source: https://github.com/libvips/libvips/issues/59#issuecomment-558997727

But I've also noticed that there is a method named `draw_smudge` which seems to do what I need, is there a reason not to use it? The comment quoted above (source) says one shouldn't use it, though.

There is also `draw_mask` which I'm guessing could be used?

Happy to hear your thoughts!

**Comment by jcupitt** [comment]
Hiya,

`draw_smudge` will force the entire image into memory, so it can be very slow. The code you listed should be a lot quicker.

You can do it a little more simply now:

```python
redact = image.gaussblur(10).crop(10, 10, 100, 100)
image = image.composite2(redact, "over", x=10, y=10)
```

ie. blur and crop out the bit you want, then composite that back on top of the original.

**Reply by sandstrom** [reply]
Awesome @jcupitt, sounds great! Thanks so much! 😄 

It's really an amazing library that you've built! 💯
